utils/pubsub/pubsub.go

```diff
@@ -15,6 +17,7 @@ type DataChannel <-chan DataEvent
 
 // PublishSubscriber stores the information about subscribers interested for a particular topic
 type PublishSubscriber struct {
+	noCopy.CopyRestrictor
```

Comment:
We are already using a `sync.RWMutex` in PublishSubscriber. Embedding a `noCopy.CopyRestrictor` appears to be redundant, since go vet will give an error for `sync.RWMutex` even if we don't use `noCopy.CopyRestrictor`.

This is was what I was trying to pinpoint in my [previous comment](https://github.com/rudderlabs/rudder-server/pull/2114#issuecomment-1168214138) as well

Reply:
This is indeed redundant for `PublishSubscriber` for the obvious reason that `sync.RWMutex` internally uses `NoCopy` struct. The whole motive of this is to start using the `CopyRestrictor` if we are copying some struct which has say a map or slice, then we might not want it to be copied or be careful at that time. If that struct have `CopyRestrictor` then govet will throw the warning. I am not able to recall a direct use-case but I am sure I have faced something similar earlier in rudder-server code.

Reply:
Fine, but let's introduce it when the need arises, wdyt?

Reply:
If someone else will keep this in mind then it is fine by me. I though I should introduce it before moving out to other team. I just want that developer's time should not be wasted by such silly mistakes. :)

Reply:
I didn't notice that the linter was complaining anyway when copying the `PublishSubscriber` struct, I took for granted it wasn't and we were trying to address that. Given that it does I completely agree with @atzoum

Reply:
@kuldeep0020 perhaps instead of throwing away the PR you know of some other code that should not be copied and that it is not using mutexes?